doc/userguide.rst

```diff
@@ -1,259 +1,1509 @@
+
+JPype Use Cases
+===============
+
+Here are three typical reasonse to use JPype.
+
+- Access to a Java library from a Python program (Python oriented)
+- Visualization of Java data structures (Java oriented)
+- Interactive Java and Python development including scientific and mathematical
+  programming.
+
+Let's explore each of these options.
+
+Case 1: Access to a Java library
+--------------------------------
+
+Suppose you are a hard core Python programmer.  You can easily use lambdas,
+threading, dictionary hacking, monkey patching, been there, done that.  You are
+hard at work on your latest project but you just need to pip in the database
+driver for your customers database and you can call it a night.  Unfortunately,
+it appears that your customers database will not connect to the Python database
+API.  The whole thing is custom and the customer isn't going to supply you with
+a Python version.  They did sent you a Java driver for the database but fat
+lot of good that will do for you.
+
+Stumbling through the internet you find a module that says it can natively
+load Java packages as Python modules.  Well, it worth a shot...
+
+So first thing the guide says is that you need to install Java and set up
+a ``JAVA_HOME`` environment variable pointing to the JRE.  Then start the
+JVM with classpath pointed to customers jar file. The customer sent over
+an example in Java so you just have to port it into Python.
+
+.. code-block:: java
+
+  package com.paying.customer;
+
+  import com.paying.customer.DataBase
+
+  public class MyExample {
+     public void main(String[] args) {
+       Database db = new Database("our_records");
+       try (DatabaseConnection c = db.connect())
+       {
+          c.runQuery();
+          while (c.hasRecords())
+          {
+            Record record = db.nextRecord();
+            ...
+          }
+       }
+    }
+  }
+
 
-Overview
---------
+It does not look too horrible to translate.  You just need to look past all
+those pointless type declarations and meaningless braces.  Once you do, you
+can glue this into Python and get back to what you really love, like performing
+dictionary comprehensions on multiple keys.
 
-JPype is an effort to allow Python programs full access to Java class
-libraries. This is achieved not through re-implementing Python, as
-Jython/JPython has done, but rather through interfacing at the native level
-in both virtual machines.
+You glance over the JPype quick start guide.  It has a few useful patterns...
+set the class path, start the JVM, remove all the type declarations, and you are done.
+
+.. code-block:: python
 
-Eventually, it should be possible to replace Java with Python in many, though
-not all, situations. JSP, Servlets, RMI servers and IDE plugins are all good
-candidates.
+   # Boiler plate stuff to start the module
+   import jpype
+   import jpype.imports
+   from jpype.types import *
 
+   # Launch the JVM
+   jpype.startJVM(classpath=['jars/database.jar'])
```

Comment:
shouldn't we pass convertStrings here in order to avoid a UserWarning?

Reply:
Or keep it as simple as possible?

Reply:
The guide was written assuming that we were releasing a 0.8 so some of the deprecated behavior was being removed.